mithril-aggregator/src/services/prover.rs

```diff
@@ -114,38 +109,51 @@ impl ProverService for MithrilProverService {
         up_to: &CardanoDbBeacon,
         transaction_hashes: &[TransactionHash],
     ) -> StdResult<Vec<CardanoTransactionsSetProof>> {
-        // 1 - Get transactions to prove per block range
-        let transactions_to_prove = self
-            .get_transactions_by_hashes_with_block_range(transaction_hashes.to_vec())
+        // 1 - Compute the set of block ranges with transactions to prove
+        let block_ranges_transactions = self.get_block_ranges(transaction_hashes).await?;
+        let block_range_transactions = self
+            .get_all_transactions_for_block_ranges(&block_ranges_transactions)
             .await?;
```

Comment:
Is that too soon or can we ask a little more from the repository by adding a method that can do both operations at the same time ?

Something like: 
```rust
    /// Get the list all transactions which are in the same block ranges that the transaction with hash provided.
    async fn get_all_transactions_inside_same_block_range(
        &self,
        hashes: Vec<TransactionHash>,
    ) -> StdResult<HashMap<BlockRange, Vec<CardanoTransaction>>>;
```

Reply:
Maybe we can leave that for the upcoming optimization phase